### PR TITLE
remove momentjs entirely with chartjs3

### PR DESCRIPTION
**Pull request by acourdavault**
## Description

Attempt to use chartjs 3 (alpha), the reason is that Chartjs 3 can work without moment entirely.

doc: https://www.chartjs.org/docs/master/getting-started/v3-migration/

image

after this moment is only in dev dependencies
image

## Type of Change
- Non breaking change ('update', 'change')

## do not merge yet

This uses version v3 alpha 1.
Since recently, Chartjs finished Alpha 2. (see here https://github.com/chartjs/Chart.js/releases/tag/v3.0.0-alpha.2)

### import changes
This version changes the import strategy to improve tree shaking.
It will require to upgrade chartjs-adapter-luxon (which currently has a non released but merged change to support this https://github.com/chartjs/chartjs-adapter-luxon/pull/20)

### CODE REVIEW

File: packages/manager/src/components/LineGraph/LineGraph.tsx [hunk 1]
```diff
@@ -242,6 +215,7 @@ const LineGraph: React.FC<CombinedProps> = (props: CombinedProps) => {
         chartInstance.current.destroy();
         chartInstance.current = null;
       }
+      // console.warn(inputEl.current, inputEl.current.getContext('2d'))
```
Comment (DevDW) [inline packages/manager/src/components/LineGraph/LineGraph.tsx:218]: Can remove

File: packages/manager/src/utilities/charts.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
-import { defaults as chartDefaults } from 'chart.js';
+// Here i do `import * as Chart` instead of `import Chart` because of issues with jest
+// jest not be able to access the namespce variable correctly
+import Chart from 'chart.js';
```
Comment (DevDW) [inline packages/manager/src/utilities/charts.ts:3]: *namespace

Should it be `import * as Chart` (stated in the comment) or `import Chart from 'chart.js'` as it is in the code?

Comment (acourdavault) [reply packages/manager/src/utilities/charts.ts:3]: I need to remove the comment, thanks

**Review by Jskobos — DISMISSED** [review]
Seems to work. Do we know when this is going to move to beta?

**Comment by acourdavault** [comment]
@johnwcallahan 
At this point yes, there is no types for this. they may come as we reach a Beta/RC status

@Jskobos 
I do not know when they plan on releasing the beta or RC

**Review by johnwcallahan — COMMENTED** [review]
The graphs definitely look a lot better; I like the fonts.

I noticed one regression where there's a discrepancy between the units/values on the graph and the units/values in the tooltip (if the scale of the graph is not the default). Example:

**Beta/Production**
Screen Shot 2020-12-01 at 2 05 40 PM


**This branch**
Screen Shot 2020-12-01 at 2 05 33 PM

In some places the value is correct the unit isn't. In other places (like above) both are wrong.

**Review by DevDW — COMMENTED** [review]
I'm also observing some disagreements between the units stated in the chart title vs. what's displayed in the tooltip. Another example where this is occurring is the Network graph on a Longview's Overview tab (`Kb/s` vs. `b/s` in tooltip)

File: packages/manager/src/components/LineGraph/LineGraph.tsx [hunk 3]
```diff
@@ -366,13 +367,27 @@ export const metricsBySection = (data: Metrics): number[] => [
...
+   * than the overall graph (if a series has a small value
+   * relative to the graph max, it will likely be rounded to
+   * zero).
+   *
+   * We need to access the raw unformatted data (passed as props.data)
+   * and apply a custom tooltip formatting function here.
+   */
+  const idx = ctx.datasetIndex;
+  const pointIdx = ctx.dataIndex;
+  const rawValue = data[idx]?.['data'][pointIdx]?.[1] ?? 0;
```
Comment (johnwcallahan) [inline packages/manager/src/components/LineGraph/LineGraph.tsx:385]: Should `[pointIdx]` be safe-accessed too, just in case?

**Review by johnwcallahan — APPROVED** [review]
Nice job @Jskobos, all the graphs I've checked against production (Linodes, NBs, Longview, Managed) match. I don't see any regressions; things are working well.

**Review by DevDW — APPROVED** [review]

**Pull request merged**